commands.py

```diff
@@ -0,0 +1,235 @@
+
+    def run(self):
+        self.env.shutdown()
+
+
+class EnsimeShowErrors(EnsimeWindowCommand):
+    def is_enabled(self):
+        return bool(self.env and
+                    self.env.is_connected() and
+                    len(self.env.window.views()) > 0 and
+                    self.env.client.analyzer_ready)
+
+    def run(self):
+        self.env.editor.show_errors = True
+        self.env.editor.redraw_all_highlights()
+
+
+class EnsimeEventListener(sublime_plugin.EventListener):
+    def on_load(self, view):
+        file = view.file_name()
+        if not (Util.is_scala(file) or Util.is_java(file)):
+            return
+        env = getEnvironment(view.window())
+        if env and env.is_connected() and env.client.analyzer_ready:
+            TypeCheckFilesReq([view.file_name()]).run_in(env, async=True)
+
+    def on_post_save(self, view):
+        file = view.file_name()
+        if not (Util.is_scala(file) or Util.is_java(file)):
+            return
+        env = getEnvironment(view.window())
+        if env and env.is_connected() and env.client.analyzer_ready:
+            TypeCheckFilesReq([view.file_name()]).run_in(env, async=True)
+
+    def on_query_completions(self, view, prefix, locations):
+        file = view.file_name()
+        if not (Util.is_scala(file) or Util.is_java(file)):
+            return
+        env = getEnvironment(view.window())
+        if env and env.is_connected() and env.client.indexer_ready:
+            if (env.editor.ignore_prefix and prefix.startswith(env.editor.ignore_prefix)):
+                return []
+            else:
+                env.editor.ignore_prefix = None
+
+            if (env.editor.current_prefix and prefix == env.editor.current_prefix):
+                env.editor.current_prefix = None
+                if view.is_popup_visible():
+                    view.hide_popup()
+                env.logger.info("Search for more suggestions either completed or was cancelled.")
+                return env.editor.suggestions
+
+            contents = (view.substr(sublime.Region(0, view.size())) if view.is_dirty()
+                        else None)
+            response = CompletionsReq(locations[0],
+                                      view.file_name(),
+                                      contents,
+                                      max_results=5).run_in(env, async=False)
+
+            if response is None:
+                return ([],
+                        sublime.INHIBIT_WORD_COMPLETIONS | sublime.INHIBIT_EXPLICIT_COMPLETIONS)
+                env.editor.ignore_prefix = prefix
+            else:
+                if len(env.editor.suggestions) > 1:
+                    CompletionsReq(locations[0], view.file_name(), contents).run_in(env, async=True)
+                    view.show_popup("Please wait while we query for more suggestions.",
+                                    sublime.HIDE_ON_MOUSE_MOVE | sublime.COOPERATE_WITH_AUTO_COMPLETE)
```

Comment:
Also, when I use one of the 5 suggestions, it does not focus on the parameters like before (where I can hit `tab` to jump and fill in the parameters).